File: CH01_CodeSmells/Solutions/DataClumps.cs
namespace RefactoringToDesignPatterns.CH01_CodeSmells.Solutions
{
    public class EventDetails
    {
        public string EventName { get; set; }
        public DateTime EventDate { get; set; }
        public string Location { get; set; }

        public void RegisterEvent(EventDetails details)
        {
            Console.WriteLine($"Event: {details.EventName}, Date: {details.EventDate}, Location: {details.Location}");
        }
    }
}

File: solutions_python/CH01_CodeSmells/DataClumps.py
from datetime import datetime
from dataclasses import dataclass
""" csharp
public void RegisterEvent(string eventName, DateTime eventDate, string location)
{
    Console.WriteLine($"Event: {eventName}, Date: {eventDate}, Location: {location}");
}
"""
# dataclass representing event details
@dataclass
class EventDetails:
    event_name: str
    event_date: datetime
    location: str


def register_event(event: EventDetails) -> str:
    return (
        f"Event: {event.event_name}"
        f"Date: {event.event_date}"
        f"Location: {event.location}"
    )


